examples/lighting-app/silabs/src/AppTask.cpp

```diff
@@ -175,6 +177,60 @@ void AppTask::LightActionEventHandler(AppEvent * aEvent)
     }
 }
 
+#ifdef SL_MATTER_RGB_LED_ENABLED
+void AppTask::LightControlEventHandler(AppEvent * aEvent)
+{
+    uint8_t light_action = aEvent->LightControlEvent.Action;
+    uint32_t value       = *reinterpret_cast<uint32_t *>(aEvent->LightControlEvent.Value);
+    uint8_t brightness   = 1;
+    PlatformMgr().LockChipStack();
+    Protocols::InteractionModel::Status status;
+    app::DataModel::Nullable<uint8_t> currentlevel;
+    // Read currentlevel value
+    status = Clusters::LevelControl::Attributes::CurrentLevel::Get(1, currentlevel);
+    PlatformMgr().UnlockChipStack();
+    if (status == Protocols::InteractionModel::Status::Success && !currentlevel.IsNull())
+    {
+        brightness = currentlevel.Value();
+    }
+    SILABS_LOG("brightness : %d ", brightness);
+    switch (light_action)
+    {
+    case LightingManager::COLOR_ACTION_XY: {
+        RgbColor_t rgb;
+        XyColor_t xy = *reinterpret_cast<XyColor_t *>(&value);
+        rgb          = XYToRgb(brightness, xy.x, xy.y);
+        sLightLED.SetColor(rgb.r, rgb.g, rgb.b);
+        ChipLogProgress(Zcl, "XY to RGB: X: %u, Y: %u | R: %u, G: %u, B: %u", xy.x, xy.y, rgb.r, rgb.g, rgb.b);
```

Comment:
sLightLED.SetColor and log can be factored out after the switch case since all case calls it

Reply:
Removed this log as part of cleanup and added it in LEDwidget APIs